getting_started/intel_debug/introduction.rst

```diff
@@ -117,6 +120,46 @@ Linux kernel to query whether or not the firmware authentication is enabled,
 which means `dmesg` logs cannot be provided to alert the user to an ME
 configuration issue.
 
+.. _loadable-libraries:
+
+1.2. Loadable libraries
+-----------------------
+
+An IPC4 library is a container of a single or multiple modules (bundle) which
+can be loaded to the firmware after it is booted up.
+Library loading is  supported on Meteor Lake (ACE1) or newer platforms.
+
+Background information: the base firmware always resides in DSP SRAM while the
+loaded library is stored in DRAM memory and only the needed code is copied to
+SRAM for execution. By moving modules out from the base firmware to a library
+can reduce the overall SRAM use depending on the device configuration and
+topology.
+
+See :ref:`llext_modules` for technical details.
+
+1.3. Monolithic and modular SOF releases
+----------------------------------------
+
+SOF project releases for Intel platforms are either monolithic (only a single firmware binary) or modular (base firmware and external libraries).
+
+1.3.1. Modular SOF releases
+---------------------------
+
+See :ref:`loadable-libraries` for details about library support in general.
+
+The released libraries are:
+ - **sof-PLAT-openmodules.ri** : the bundle contains modules for audio processing not included in the base firmware
+ - **sof-PLAT-debug.ri** : the bundle contains modules that are needed for firmware debugging and profiling. Used by developers and for bug reporting if needed
+ - **UUID.bin** : Mainly 3rd party libraries identified by UUID. If the library contains multiple modules then a UUID symlink must be provided for each one.
+
+Notes:
+ - The Kernel will attempt to load \*-openmodules.ri followed by \*-debug.ri from the library path after the base firmware boot if they exist.
```

Comment:
In case of modular SOF release the basefw and modules are in different libraries (note that the basefw itself is a library file).
The full SOF is basefw+openmodules+debug, this is what we have now as monolithic single basefw, it includes all modules.
Iow, to get the full SOF you need to boot the basefw and then load the modules from the external libraries.

The UUID based loading is still supported, but the _full_ SOF is defined in case of modular release as basefw+openmodules+debug.

Reply:
Ok. And all those three base-release libraries are loaded to SRAM at boot time, like the the older monolithic FW binary, no?

Reply:
The basefw loaded (to SRAM) and booted, then we load the openmodules followed by the debug library, these will be loaded to DRAM. The modules that are needed for the audio stream will be loaded (by the firmware) from DRAM to SRAM for execution.
The monolithic firmware with all the modules are stored in SRAM.

The basefw is still special, it is the one that can boot and have all the needed infra (like the kernel in Linux). Instead of individual modules we can have module bundles.

At the end these form the SOF package either way: monolithic or modular.

Reply:
I finally read the long mail thread about the subject from just before my vacation, and it answered some of my questions. But still, the modules in the open-modules-bundle are loaded as needed based on uuid's even if they are not needed for getting access to the library file itself, don't they?

Reply:
No, sof-bin release is not planned to have UUID symlinks to the openmodules.ri, it is going to be loaded unconditionally if it exists in filesystem.
We can only load library into firmware, the library can contain one or more modules (bundle). One can freely move modules between basefw and openmodules between releases, the _whole_ package will contain all the needed modules.

I want to define this as simple as possible:
If sof-bin is **monolithic** for a platform then **only** the **basefw** will be on filesystem.
If sof-bin is **modular** for a platform then **basefw+openmodules(+debug)** will be on the filesystem.

The kernel shall load these unconditionally, which means that the kernel will treat the base libraries as optional, if they don't exist it is not an error since it is the monolithic release.

The kernel change will come after the agreement on the documentation and how the sof-bin is going to be released.

The kernel will still going to look based on UUID for missing modules but for a complete sof-bin release it is not really going to happen (apart from LNL which have one module outside as library).